src/Tasks/Microsoft.NET.Build.Tasks/GenerateResxCodeBehindFile.cs

```diff
@@ -0,0 +1,233 @@
+            builder.AppendLine("using System.Reflection");
+            builder.AppendLine();
+
+            string resourceName = resxFile.GetMetadata("ManifestResourceName");
+            string[] nameParts = resourceName.Split('.');
+            string namespaceName, className, classIndent;
+            if (nameParts.Length > 1)
+            {
+                namespaceName = string.Join(".", nameParts, 0, nameParts.Length - 1);
+                className = nameParts.Last();
+                classIndent = "    ";
+            }
+            else
+            {
+                namespaceName = null;
+                className = nameParts[0];
+                classIndent = "";
+            }
+
+            string memberIndent = classIndent + "    ";
+
+            if (namespaceName != null)
+                builder.AppendLine($"namespace {namespaceName} {{");
+
+            builder.AppendLine($"{classIndent}internal static class {className}");
+            builder.AppendLine("{");
+
+            builder.AppendLine($"{memberIndent}internal static global::System.Globalization.CultureInfo Culture {{ get; set; }}");
+            builder.AppendLine($"{memberIndent}internal static global::System.Resources.ResourceManager ResourceManager {{ get; }} = new global::System.Resources.ResourceManager(\"{resourceName}\", typeof({className}).GetTypeInfo().Assembly);");
+            builder.AppendLine();
+
+            foreach (var node in XDocument.Load(resxFile.ItemSpec).Descendants("data"))
+            {
+                string name = node.Attribute("name")?.Value ?? throw new InvalidDataException("Missing resource name");
+                string value = node.Elements("value").FirstOrDefault()?.Value.Trim() ?? throw new InvalidDataException($"Missing resource value: '{name}'");
+                string comment = node.Elements("comment").FirstOrDefault()?.Value.Trim();
+
+                if (name == "")
+                    throw new InvalidDataException("Empty resource name");
+
+                // If the comment was provided by the developer, use that.
+                // Otherwise, generate a default comment like the old ResX generator does.
+                if (comment == null)
+                {
+                    const int maxDocCommentLength = 256;
+                    if (value.Length > maxDocCommentLength)
+                        comment = "Looks up a localized string similar to " + value.Substring(0, maxDocCommentLength) + " [truncated].";
+                    else
+                        comment = $"Looks up a localized string similar to {value}.";
+                }
+
+                string escapedTrimmedValue = new XAttribute("summary", comment).ToString();
+                foreach (var line in escapedTrimmedValue.Split(new[] { "\r\n", "\r", "\n" }, StringSplitOptions.None))
+                    builder.AppendLine($"{memberIndent}/// {line}");
+
+                string identifier = IsLetterCharacter(name[0]) ? name : "_" + name;
```

Comment:
This is only one case of an invalid identifier. I will see if we can find the code from the CodeDOM-based generator in order to get parity with it. I see that it knows to make "a b", "a_b", and even "for" into "_for".